packages/shared/src/contexts/payment/StoreKit.tsx

```diff
@@ -118,6 +150,58 @@ export const StoreKitSubProvider = ({
     [user?.subscriptionFlags?.appAccountToken],
   );
 
+  useEffect(() => {
+    const eventName = 'iap-purchase-event';
+    promisifyEventListener<void, PurchaseEvent>(
+      eventName,
+      (event) => {
+        const { name, detail, product } = event.detail;
+        switch (name) {
+          case PurchaseEventName.PurchaseCompleted:
+            logRef.current({
+              event_name: LogEvent.CompleteCheckout,
+              extra: {
+                user_id: user?.id,
+                cycle: productIds[product.attributes.offerName].duration,
+                localCost: product.attributes.offers[0].price,
+                localCurrency: product.attributes.offers[0].currencyCode,
+                payment: SubscriptionProvider.AppleStoreKit,
+              },
+            });
+            router.push(plusSuccessUrl);
+            break;
+          case PurchaseEventName.PurchaseInitiated:
+            logRef.current({
+              event_name: LogEvent.InitiatePayment,
+            });
+            break;
+          case PurchaseEventName.PurchaseFailed:
+            logRef.current({
+              event_name: LogEvent.ErrorCheckout,
+              extra: {
+                errorCode: detail,
+              },
+            });
+            displayToast(DEFAULT_ERROR);
+            break;
+          case PurchaseEventName.PurchasePending:
+            displayToast('Please wait for the purchase to be completed.');
+            break;
+          case PurchaseEventName.PurchaseCancelled:
```

Comment:
@idoshamun do you want an event for when they user cancels a purchase? I.e. clicks out of the purchase sheet.

Reply:
let's leave it for now